packages/adapter-components/src/config/ducktype.ts

```diff
@@ -75,13 +75,18 @@ export const validateApiDefinitionConfig = (
       isDefined,
     ),
   )
+  const configMap = _.pickBy(
+    _.mapValues(adapterApiConfig.types, typeDef => typeDef.transformation),
+    isDefined,
+  )
+  // TODO: remove this check once singleton types are implemented in ducktype
+  if (_.filter(configMap, ['isSingleton', true]).length > 0) {
```

Comment:
nit: I think we should not support any value in that config, not just `true`. also, you can use `.some()` instead of `.filter(...).length > 0`
```suggestion
  if (Object.values(configMap).some(def => def.isSingleton !== undefined) {
```